### PR TITLE
[devtools week] Update OWNERS file & update release section

### PR DESCRIPTION
**Pull request by thepetk**
### What does this PR do?:

* Includes all team inside OWNERS file
* Updates the release section inside README.md with a list of steps to create a new release. It also mentions explicitly that releases are created on demand.
 
### Which issue(s) this PR fixes:
task of devtools week sep 2023

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [x] Unit/Functional tests

  

- [x] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [x] Documentation 

   

- [x] Client Impact

  

- [x] Gosec scans

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -7,9 +7,19 @@ approvers:
 - maysunfaisal
 - mik-dass
```
Comment (kim-tsao) [inline OWNERS:8]: I think we can remove this ID

Comment (thepetk) [reply OWNERS]: Done :)

File: OWNERS [hunk 2]
```diff
@@ -5,11 +5,20 @@ approvers:
 - elsony
 - kim-tsao
 - maysunfaisal
-- mik-dass
 - yangcao77
+- johnmcollier
+- michael-valdron
+- thepetk
+- feloy
+- rm3l
```
Comment (maysunfaisal) [inline OWNERS:13]: do we need all these new folks on the reviewers and approvers list?

Comment (thepetk) [reply OWNERS]: updated!

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**